pkg/test/framework/label/instance.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package label
+
+// Instance is a label instance.
+type Instance string
+
+// Set is a set of labels
+type Set []Instance
+
+// NewSet returns a new label set.
+func NewSet(labels ...Instance) Set {
+	return Set(labels)
+}
+
+// Join returns a set that is concatenation of l and s.
+func (l Set) Join(s Set) Set {
```

Comment:
Given this is called a Set, the choice of the name Join seems confusing (as in my mind it implies ordering). Would Union be more apt?

Alternatively, since this type is actually not really a set since it doesn't have singleton property, then maybe the whole type should be called something else, at which point Join may make more sense.